### PR TITLE
chore(transformers): Use `client.TransformWithStruct` instead of appending default options

### PR DESCRIPTION
**Pull request by candiduslynx**
Sometimes we have default transformer options (e.g., type & resolver transformers).
We use them like this:
```go
schema.Table{
  Transform: transformers.TransformWithStruct(&someStruct, append(client.DefaultOptions, a,b,c)...),
}
```

However, it’ll be much easier to do the following instead:
- in the client package define the following func
  ```go
  func TransformWithStruct(t any, opts ...transformers.StructTransformerOption) schema.Transform {
    return transformers.TransformWithStruct(t, append(options, opts...)...)
  }
  ```
- And replace resource transform calls to `client.TransformWithStruct(t, a, b, c)`

This PR does exactly this.

### CODE REVIEW

File: plugins/source/launchdarkly/client/transformers.go [hunk 1]
```diff
@@ -4,13 +4,16 @@ import (
...
 
+	"github.com/cloudquery/plugin-sdk/schema"
 	"github.com/cloudquery/plugin-sdk/transformers"
 )
 
-func SharedTransformers(others ...transformers.StructTransformerOption) []transformers.StructTransformerOption {
-	return append([]transformers.StructTransformerOption{
-		transformers.WithNameTransformer(nameTransformer),
-	}, others...)
+var options = []transformers.StructTransformerOption{
```
Comment (erezrokah) [inline plugins/source/launchdarkly/client/transformers.go:11]: Maybe drop this also as there's only a single option?

File: plugins/source/mixpanel/client/transformers.go [hunk 2]
```diff
@@ -8,10 +8,12 @@ import (
 	"github.com/cloudquery/plugin-sdk/transformers"
 )
 
-func SharedTransformers(others ...transformers.StructTransformerOption) []transformers.StructTransformerOption {
-	return append([]transformers.StructTransformerOption{
-		transformers.WithTypeTransformer(typeTransformer),
-	}, others...)
+var options = []transformers.StructTransformerOption{
```
Comment (erezrokah) [inline plugins/source/mixpanel/client/transformers.go:11]: Same here

**Review by erezrokah — APPROVED** [review]

**Comment by erezrokah** [comment]
Non blocking but maybe drop the options where there's a single one. Can be done in a follow up

File: plugins/source/azuredevops/client/transformers.go [hunk 3]
```diff
@@ -25,10 +25,10 @@ func typeTransformer(field reflect.StructField) (schema.ValueType, error) {
 	return schema.TypeInvalid, nil
 }
 
-func Options() []transformers.StructTransformerOption {
-	options := []transformers.StructTransformerOption{
-		transformers.WithTypeTransformer(typeTransformer),
-	}
+var options = []transformers.StructTransformerOption{
```
[the remark this answers is not in the document]
Comment (candiduslynx) [reply plugins/source/azuredevops/client/transformers.go:28]: I don't have any opinion on that, but I made it universal for every plugin to be just a copy-paste funcs, and those work with slice for options.
Do you have a strong opinion on this?

**Pull request merged**